### PR TITLE
Feat/video listing

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. Video listing page
2. filter by category
3. sort by latest

### CODE REVIEW

File: src/component/video-card/VideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import './video-card.css';
+import MoreVertOutlinedIcon from '@mui/icons-material/MoreVertOutlined';
+
+const VideoCard = ({video}) => {
```
Comment (Maianki) [inline src/component/video-card/VideoCard.jsx:4]: suggestion: you can destructure this video prop further so you don't have to type video.propertyName all the time.

File: src/pages/home/index.jsx [hunk 2]
```diff
@@ -1,11 +1,66 @@
...
+          categories: categories.map(cateogry => {
+            return cateogry._id === id ? { ...cateogry, selected: true } : { ...cateogry, selected: false }
+          }),
+        }
+      });
+      setLoading(false);
+    }, 700);
+  }
+  const sortedVideos = getFilteredSotredVideos(videos, categorizedBy, sortBy);
+  console.log(sortedVideos);
```
Comment (Maianki) [inline src/pages/home/index.jsx:28]: Please remove logs

File: src/pages/home/index.jsx [hunk 3]
```diff
@@ -1,11 +1,66 @@
...
+        type: 'FILTER_BY_CATEGORY',
+        payload: {
+          categorizedBy: categoryName,
+          categories: categories.map(cateogry => {
+            return cateogry._id === id ? { ...cateogry, selected: true } : { ...cateogry, selected: false }
+          }),
+        }
+      });
+      setLoading(false);
+    }, 700);
```
Comment (Maianki) [inline src/pages/home/index.jsx:25]: query:  Why are you dispatching it with a delay?

File: src/route/ProtectedRoute.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Outlet, Navigate } from "react-router-dom";
+
+const ProtectedRoute = () => {
+  //const isAuth = localStorage.getItem("isAuth");
```
Comment (Maianki) [inline src/route/ProtectedRoute.jsx:4]: Please remove commented code if you will not use it in future PR

**Review by Maianki — APPROVED** [review]
Great job. Everything looks good has added a few minor suggestions. 

Things I liked about your code- 
- Good naming convention and folder structure.
- Your code is well abstracted and readable.
- Also, you are using BEM so great work there too.

**Review by rutvikpumak — APPROVED** [review]
Code looks good and well structred.
Mainaki already suggested some changes and do that and its good to merge

**Pull request merged**